warehouse/query-core/src/main/java/datawave/query/jexl/visitors/RegexFunctionVisitor.java

```diff
@@ -47,61 +50,56 @@ public Object visit(ASTFunctionNode node, Object data) {
-                    
-                    if (child instanceof ASTIdentifier) {
-                        this.adopt(newParent, buildRegexNode((ASTIdentifier) child, functionMetadata.name(), arguments.get(1).image));
-                    } else { // probably a Reference
-                        for (int j = 0; j < child.jjtGetNumChildren(); j++) {
-                            JexlNode maybeIdentifier = child.jjtGetChild(j);
-                            if (maybeIdentifier instanceof ASTIdentifier) {
-                                this.adopt(newParent, buildRegexNode((ASTIdentifier) maybeIdentifier, functionMetadata.name(), arguments.get(1).image));
-                            }
+            }
+            
+            // only re-parent if the same number of regex nodes were built
+            if (identifiers.size() == children.size()) {
+                switch (identifiers.size()) {
+                    case 0:
+                        return returnNode;
+                    case 1:
+                        return children.get(0);
+                    default:
+                        if (functionMetadata.name().equals(INCLUDE_REGEX)) {
+                            return JexlNodeFactory.createOrNode(children);
+                        } else {
+                            return JexlNodeFactory.createAndNode(children);
                         }
-                    }
-                }
-                if (newParent.jjtGetNumChildren() == node0.jjtGetNumChildren() && newParent.jjtGetNumChildren() != 0) {
-                    returnNode = newParent;
                 }
             }
         }
         return returnNode;
     }
     
-    private void adopt(JexlNode parent, JexlNode child) {
-        if (child != null && parent != null) {
-            child.jjtSetParent(parent);
-            parent.jjtAddChild(child, parent.jjtGetNumChildren());
-        }
-    }
-    
+    /**
+     * Builds a regex node given a field name, regex and original function name.
+     * 
+     * @param identifier
+     *            the field
+     * @param functionName
+     *            the original function name, either <code>includeRegex</code> or <code>excludeRegex</code>
+     * @param regex
+     *            the regex
+     * @return a new regex node, or null if no such regex node could be built
+     */
     private JexlNode buildRegexNode(ASTIdentifier identifier, String functionName, String regex) {
         String field = JexlASTHelper.deconstructIdentifier(identifier.image);
         if (nonEventFields.contains(field.toUpperCase())) {
             try {
                 JavaRegexAnalyzer jra = new JavaRegexAnalyzer(regex);
                 if (!jra.isNgram()) {
-                    if (functionName.equals("includeRegex")) {
-                        if (log.isDebugEnabled())
-                            log.debug("Building new ER Node");
+                    if (functionName.equals(INCLUDE_REGEX)) {
```

Comment:
Did this visitor end up being a bit too chatty with debug messages when the ER node was built? Was wondering if there is anything here that would be considered loggable even at a trace level.

Reply:
The debug message wasn't informative so I decided to just rip out all logging for this visitor. If you would like I can add in a trace level message that logs when a filter function is rewritten

Reply:
That sounds good to me, thanks.

Reply:
That sounds good to me, thanks.

Reply:
Added the logger back and put in a more informative message